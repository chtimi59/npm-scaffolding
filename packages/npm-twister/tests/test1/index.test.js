const path = require('path')
const childProcess = require('node-child-process-extension')
const fs = require('node-fs-extension')
const nodePrefix = `node ${path.resolve(__dirname, '../../npmt.js')}`
async function npmt(args="") {
    cmd = `${nodePrefix} ${args}`
    return childProcess.extras.exeToStdOut(cmd, __dirname)
}
const packageJson = path.resolve(__dirname, 'package.json')

describe('npm-twister', function () {
    describe('test1', function () {
        beforeAll(async function () {
            await fs.extras.rm(packageJson)
        })
        it('empty', async function () {
            try {
                const ret = await npmt("")
                expect(ret).toMatchSnapshot()
            } catch(e) {
                console.error(e)
                throw(e)
            }
        })
        it('local script description', async function () {
            const ret = await npmt("hello --help")
            expect(ret).toMatchSnapshot()
        })
        it('run local script', async function () {
            const ret = await npmt("hello")
            expect(ret.toUpperCase()).toEqual(packageJson.toUpperCase())
        })
        it('run npm init command', async function () {
            await npmt("init -y")
            const ret = await fs.extras.exists(packageJson, 'file')
            expect(ret).toEqual(true)
        })
        it('bump major version', async function () {
            await npmt("init -y")
            await npmt("version major")
            const ret = await npmt("getversion")
            expect(ret).toEqual("2.0.0")
        }, 15000)
    })
})
