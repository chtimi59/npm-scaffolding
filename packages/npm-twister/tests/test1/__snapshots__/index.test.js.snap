// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm-twister test1 empty 1`] = `
"Usage:  npmt <command>

wrapped commands from npm (v6.4.1):

    available:
    access, add, add-user, adduser, audit, author, bin, bugs, c, cache
    ci, completion, config, create, ddp, dedupe, deprecate, dist-tag, dist-tags, docs
    doctor, edit, explore, find, find-dupes, get, help, help-search, home, hook
    i, info, init, innit, install, isntall, issues, it, la, link
    list, ll, ln, login, logout, ls, outdated, owner, pack, ping
    prefix, profile, prune, publish, r, rb, rebuild, remove, repo, restart
    rm, root, rum, run, run-script, s, se, search, set, show
    shrinkwrap, star, stars, start, stop, t, team, test, token, tst
    udpate, un, uninstall, unlink, unpublish, unstar, up, update, upgrade, urn
    v, version, view, whoami


where:

    getversion  alias: h

    hello       Some description about hello.js
                alias: h

Specific help example:

    npmt install --help

version: 0.0.1"
`;

exports[`npm-twister test1 local script description 1`] = `"Some description about hello.js"`;
